src/components/show-products/index.js

```diff
@@ -117,50 +121,59 @@ export const ShowProducts = () => {
           />
         </label>
         {searchTerm && (
-          <input
+          <button
             className="close-icon"
-            onClick={() => setSearchTerm('')}
-            type="reset"
-            value="X"
             aria-label="This button clears the content of the search field."
-          />
+            onClick={() => setSearchTerm('')}
+          >
+            <FiX />
+          </button>
         )}
       </form>
 
       {list.length > 0 ? (
-        <ul>
-          {list.map(({ id, productName, lastPurchaseDate, timeFrameLabel }) => (
-            <li className="checkbox-item" key={id}>
-              <label htmlFor={id} className="checkbox-label">
-                <input
-                  type="checkbox"
-                  id={id}
-                  name={productName}
-                  onChange={(event) => handleCheckboxChange(event, id)}
-                  checked={
-                    lastPurchaseDate &&
-                    diffBetweenTodayAndDate(lastPurchaseDate.toDate()) < one_day
+        <ul className="products-list">
+          {list.map(({ id, productName, lastPurchaseDate, timeFrameLabel }) => {
+            const checked =
+              lastPurchaseDate &&
+              diffBetweenTodayAndDate(lastPurchaseDate.toDate()) < one_day;
+
+            return (
+              <li
+                className={`checkbox-item checkbox-timeFrame-${timeFrameLabel}`}
+                key={id}
+              >
+                <label
+                  htmlFor={id}
+                  className={
+                    checked ? 'checkbox-label checked-label' : 'checkbox-label'
                   }
-                  aria-label={TimeFrameLabels[timeFrameLabel]}
-                />
-                <span
-                  className={`checkmark checkbox-timeFrame-${timeFrameLabel}`}
-                ></span>
-                <span className="checkbox-name">{productName}</span>
+                >
+                  <input
+                    type="checkbox"
+                    id={id}
+                    name={productName}
+                    onChange={(event) => handleCheckboxChange(event, id)}
+                    checked={checked}
+                    aria-label={TimeFrameLabels[timeFrameLabel]}
+                  />
```

Comment:
I could identify some issues when you try to add a product with a long name. Please, consider adding a [`maxLength` property](https://developer.mozilla.org/es/docs/Web/HTML/Element/input) to your `input`.

![image](https://user-images.githubusercontent.com/8689897/142735487-4699b580-3446-4585-866d-5c90312727e9.png)

Reply:
Done :)